### PR TITLE
Dartdoc test -- full diff

### PR DESCRIPTION
**Pull request by jonasfj**
Creating a PR with all dartdoc_test related changes for review.

### CODE REVIEW

File: dartdoc_test/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1 @@
+## v0.1.0
```
Comment (jonasfj) [inline dartdoc_test/CHANGELOG.md:1]: We should write a proper changelog entry :D

File: dartdoc_test/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+.dartdoc_test
```
Comment (jonasfj) [inline dartdoc_test/.gitignore:1]: Do we still create this, or do we create things inside `.dart_tool/dartdoc_test/`

File: dartdoc_test/lib/dartdoc_test.dart [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  for (final r in result) {
+    for (final e in r.errors) {
+      reporter.addIssue(Issue(
+        path: r.file.sample.comment.path,
+        message: e.error.message,
+        commentSpan: e.commentSpan,
+        generatedSpan: e.generatedSpan,
+      ));
+    }
+  }
```
Comment (chinmoy12c) [inline dartdoc_test/lib/dartdoc_test.dart:57]: Would it be good to add an example of how the reporter represents the result?

File: dartdoc_test/lib/src/analyzer.dart [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  if (offset == -1) {
+    return null;
+  }
+  return original.subspan(offset, offset + sample.length);
+}
+
+class DartdocErrorResult {
+  final AnalysisError error;
+  final FileSpan? commentSpan;
+  final FileSpan? generatedSpan;
```
Comment (chinmoy12c) [inline dartdoc_test/lib/src/analyzer.dart:62]: Maybe add one-liner documentation for these properties?

**Pull request merged**